AM43 component can’t find sensor.h

### The problem

AM43 blinds component won’t compile.

Tried with `  libraries:
    - "Sensor@0.1.0" ` and without.

### Which version of ESPHome has the issue?

2021.12.1

### What type of installation are you using?

pip

### Which version of Home Assistant has the issue?

_No response_

### What platform are you using?

ESP32

### Board

nodemcu

### Component causing the issue

am43

### Example YAML snippet

```yaml
esphome:
  name: kitchenblinds
  platform: ESP32
  board: nodemcu-32s
  libraries:
    - "Sensor@0.1.0"

wifi:
  ssid: "SSID"
  password: "Password"

captive_portal:

# Enable logging
logger:

# Enable Home Assistant API
api:
  password: "Password"

ota:
  password: "Password"

web_server:
  port: 80

esp32_ble_tracker:

ble_client:
  - mac_address: AA:BB:CC:DD:EE:FF
    id: am43_kitchen

cover:
  - platform: am43
    name: "Kitchen Blinds"
    ble_client_id: am43_kitchen
```


### Anything in the logs that might be useful for us?

```txt
Library Manager: ArduinoJson-esphomelib @ 5.13.3 has been installed!
Dependency Graph
|-- <AsyncTCP-esphome> 1.2.2
|-- <Sensor> 0.1.0
|-- <WiFi> 1.0
|-- <FS> 1.0
|-- <Update> 1.0
|-- <ESPAsyncWebServer-esphome> 2.1.0
|   |-- <AsyncTCP-esphome> 1.2.2
|-- <DNSServer> 1.1.0
|-- <ArduinoJson-esphomelib> 5.13.3
|-- <ESPmDNS> 1.0
Compiling .pioenvs/kitchenblinds/src/esphome/components/am43/am43.cpp.o
Compiling .pioenvs/kitchenblinds/src/esphome/components/am43/am43_base.cpp.o
Compiling .pioenvs/kitchenblinds/src/esphome/components/am43/cover/am43_cover.cpp.o
In file included from src/esphome/components/am43/am43.cpp:1:0:
src/esphome/components/am43/am43.h:6:46: fatal error: esphome/components/sensor/sensor.h: No such file or directory
compilation terminated.
*** [.pioenvs/kitchenblinds/src/esphome/components/am43/am43.cpp.o] Error 1
Compiling .pioenvs/kitchenblinds/src/esphome/components/api/api_connection.cpp.o
========================================================================= [FAILED] Took 9.42 seconds =========================================================================
```


### Additional information

_No response_

--- a/esphome/components/am43/cover/__init__.py
+++ b/esphome/components/am43/cover/__init__.py
@@ -5,7 +5,7 @@ from esphome.const import CONF_ID, CONF_PIN
 
 CODEOWNERS = ["@buxtronix"]
 DEPENDENCIES = ["ble_client"]
-AUTO_LOAD = ["am43"]
+AUTO_LOAD = ["am43", "sensor"]
 
 CONF_INVERT_POSITION = "invert_position"
 

